cmd/shadowsocks-local/local.go

```diff
@@ -347,10 +350,61 @@ func enoughOptions(config *ss.Config) bool {
 		config.LocalPort != 0 && config.Password != ""
 }
 
+func parseURI(u string, cfg *ss.Config) (string, error) {
+	if u == "" {
+		return "", nil
+	}
+	invalidURI := errors.New("invalid URI")
+	// ss://base64(method:password)@host:port
+	// ss://base64(method:password@host:port)
```

Comment:
What if `@` occours in password? 
Do we need to escape characters?

Reply:
URI generated by shadowsocks-android is `ss://base64(method:password)@host:port`
`method:password` was encoded into base64 string.
We only need to deal with `:` in password. As far as I know, there is no crypto methods whose name contains `:`, so I use `bytes.SlpitN(dat, sub, 2)` to split `method:password` into 2 parts
I didn't have a look on URI generated by other clients.

And the pattern `ss://base64(method:password@host:port)`, I found it from a website which provides free ss configs with simple password.

Reply:
Great,

Awesome Work